File: Plugins/Aspose_Words_Java_for_Ruby/lib/asposewordsjavaforruby/simplemailmerge.rb
module Asposewordsjavaforruby
  module SimpleMailMerge
    def initialize()
        mail_merge()
    end
        
    def mail_merge()
        # The path to the documents directory.
        data_dir = File.dirname(File.dirname(File.dirname(__FILE__))) + '/data/quickstart/'

        # Open the document.
        doc = Rjb::import('com.aspose.words.Document').new(data_dir + "MailMerge.doc")
        # Fill the fields in the document with user data.
        doc.getMailMerge().execute(
            Array["FullName", "Company", "Address", "Address2", "City"],
            Array["James Bond", "MI5 Headquarters", "Milbank", "", "London"]
        )
        # Saves the document to disk.
        doc.save(data_dir + "MailMerge Out.docx")
    end 

  end
end

File: Examples/src/main/java/com/aspose/words/examples/mail_merge/SimpleMailMerge.java
package com.aspose.words.examples.mail_merge;

import com.aspose.words.Document;
import com.aspose.words.examples.Utils;


public class SimpleMailMerge {
    public static void main(String[] args) throws Exception {
        //ExStart:1
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(SimpleMailMerge.class);
        // Open the document.
        Document doc = new Document(dataDir + "MailMerge.ExecuteArray.doc");
        doc.getMailMerge().setUseNonMergeFields(true);
        doc.getMailMerge().execute(
                new String[]{"FullName", "Company", "Address", "Address2", "City"},
                new Object[]{"James Bond", "MI5 Headquarters", "Milbank", "", "London"});
        // Save the output document to disk.
        doc.save(dataDir + "Output.doc");
        //ExEnd:1
    }
}